qiita_db/base.py

```diff
@@ -184,47 +192,60 @@ class QiitaStatusObject(QiitaObject):
     _status_setter_checks
     """
 
-    @property
-    def status(self):
-        r"""String with the current status of the analysis"""
-        # Check that self._table is actually defined
-        self._check_subclass()
+    def status(self, trans=None):
+        r"""String with the current status of the object
 
+        Parameters
+        ----------
+        trans: Transaction, optional
+            Transaction in which this method should be executed
+        """
         # Get the DB status of the object
-        conn_handler = SQLConnectionHandler()
-        return conn_handler.execute_fetchone(
-            "SELECT status FROM qiita.{0}_status WHERE {0}_status_id = "
-            "(SELECT {0}_status_id FROM qiita.{0} WHERE "
-            "{0}_id = %s)".format(self._table),
-            (self._id, ))[0]
-
-    def _status_setter_checks(self, conn_handler):
+        trans = trans if trans is not None else Transaction(
```

Comment:
...though the comment above can be generalized slightly further to allow passing in a name or something